### PR TITLE
Fix #8106: FilterMatchMode.IN check for empty string

### LINKED ISSUE #8106: FilterMatchMode.IN : allow to filter null values
[primefaces-test-master.zip](https://github.com/primefaces/primefaces/files/7590646/primefaces-test-master.zip)
**Description of the defect**
filterMatchMode="in" doesn't properly filter value of null. 
This issue is similar to https://github.com/primefaces/primeng/issues/5735 
* Column has empty cells (null cells)
* Filter selectCheckboxMenu has <f:selectItem itemValue="" itemLabel="(Empty)"/>
* If this (Empty) filter item is selected, cells were not reduced to only empty cells.

**Reproducer**
ZIP is attached.

**Environment:**
 - PF Version: _11.0.0-RC2
 - MyFaces 2.3
 - Affected browsers:  Edge

**To Reproduce**
1. Run demo with mvn jetty:run
2. Filter table by column 'Companies' - select '(Empty)' item in selectCheckboxMenu
3. All rows are still there, but there should be only two rows, where company cells are empty

**Expected behavior**
When `<f:selectItem itemValue="" itemLabel="(Empty)"/>` is selected in selectCheckboxMenu filter, empty cells in this column should remain.  

**Example XHTML**
```html
<h:form>
        <p:dataTable var="customer" value="#{dtFilterView.customers1}" widgetVar="customersTable"
                     emptyMessage="No customers found with given criteria"
                     filteredValue="#{dtFilterView.filteredCustomers1}">

            <p:column field="company" headerText="Companies" sortBy="#{customer.company}" filterBy="#{customer.company}"
                      filterMatchMode="in">
                <f:facet name="filter">
                    <p:selectCheckboxMenu onchange="PF('customersTable').filter()"
                                          label="Select"
                                          value="#{dtFilterView.selectedCompanies}" >
                        <f:selectItem itemValue="" itemLabel="(Empty)"/>
                        <f:selectItems value="#{dtFilterView.companies}" />
                    </p:selectCheckboxMenu>
                </f:facet>
            </p:column>

        </p:dataTable>
    </h:form>
```

### PR DESCRIPTION
**Pull request by melloware**
Tested with reproducer and its working as expected.

### CODE REVIEW

**Review by alinatoleeva — COMMENTED** [review]
Looks great. Thanks for the very quick response and the test.

File: primefaces/src/main/java/org/primefaces/model/filter/InFilterConstraint.java [hunk 1]
```diff
@@ -49,6 +50,11 @@ else if (filter instanceof Collection) {
             if (Objects.equals(value, o)) {
                 return true;
             }
+
+            // GitHub #8106 check for "" comparison
+            if (o instanceof String && o.toString().length() == 0 && value == null) {
```
Comment (tandraschko) [inline primefaces/src/main/java/org/primefaces/model/filter/InFilterConstraint.java]: can o.toString return null?
maybe better user LangUtils.isEmpty(o.toString())

**Comment by alinatoleeva** [comment]
I found out, that with this change the clearFilters() function does not work anymore. 
```
if (PrimeFaces.widget.DataTable) {
	PrimeFaces.widget.DataTable.prototype.clearFilters = function() {
		this.thead.find('> tr > th.ui-filter-column > .ui-column-filter').val('');
		this.thead.find('> tr > th.ui-filter-column > .ui-column-customfilter :input').val(''); //this line
		console.log(undefined)
		$(this.jqId + '\\:globalFilter').val('');
	};
}
```

**this.thead.find('> tr > th.ui-filter-column > .ui-column-customfilter :input').val('')** is setting dropdown's value to empty string and that's why the  **<f:selectItem itemValue="" itemLabel="(Empty)"/>** is getting selected when the clearFilters is called, but expected, that all items in dropdown have to be deselected

**Comment by tandraschko** [comment]
https://github.com/primefaces/primefaces/issues/8071

clearFilters needs to be enhanced for selectCheckboxMenu
lets then check the issue again

**Comment by tandraschko** [comment]
@alinatoleeva 
your issue with clearFilters is unrelated to this issue actually?

**Comment by alinatoleeva** [comment]
> @alinatoleeva your issue with clearFilters is unrelated to this issue actually?

@tandraschko clearFilters has stoped to work correctly when I icluded this fix (#8107) in my project. 
But then I added additional fix for selectCheckboxMenu from here: https://github.com/primefaces/primefaces/issues/8104#issuecomment-976456072  and it works again.

File: primefaces/src/main/java/org/primefaces/model/filter/InFilterConstraint.java (hunk 1 above)
Comment (melloware) [reply primefaces/src/main/java/org/primefaces/model/filter/InFilterConstraint.java]: Yep switched to Langutils!  Good Catch

**Comment by melloware** [comment]
@alinatoleeva just confirming your real issue had nothing to do with this fix correct?

**Comment by alinatoleeva** [comment]
> @alinatoleeva just confirming your real issue had nothing to do with this fix correct?

@melloware my real issue is #8106 . So #8107 fixed it. 
clearFilters has nothing to do with fix #8107, correct.

**Pull request merged**